src/Scorer.js

```diff
@@ -0,0 +1,41 @@
+var Scorer = function() {
+  this.rolls = [];
+  this.frameCounter = 0;
+};
+
+Scorer.prototype.roll = function(score) {
+  this.rolls.push(score);
+  if (this.frameCounter <10) {
+    if (score === 10) {
+      this.frameCounter +=1;}
+    else {
+      this.frameCounter +=0.5;};}
```

Comment:
Interesting approach! Good that it works, but be careful with floats, they can sometimes fail in sneaky ways